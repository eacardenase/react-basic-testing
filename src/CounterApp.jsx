import { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value }) => {
    const [counter, setCounter] = useState(value);

    const incrementCounter = () => {
        // setCounter(counter + 1);
        setCounter((c) => c + 1);
    };

    const decrementCounter = () => setCounter((c) => c - 1);

    const resetCounter = () => setCounter(value);

    return (
        <>
            <h1>CounterApp</h1>
            <h2> {counter} </h2>

            <button onClick={decrementCounter}>-1</button>
            <button onClick={resetCounter}>Reset</button>
            <button onClick={incrementCounter}>+1</button>
        </>
    );
};

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
};

export default CounterApp;
