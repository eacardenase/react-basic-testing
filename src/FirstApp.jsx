import PropTypes from 'prop-types';

FirstApp.propTypes = {
    subtitle: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
    subtitle: "I'm a software developer",
    // title: 'John Doe',
};

export function FirstApp({ title, subtitle, name }) {
    return (
        <>
            <h1> {title} </h1>
            <p>{subtitle}</p>
            <p>{name}</p>
        </>
    );
}
