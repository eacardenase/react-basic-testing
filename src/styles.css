html,
body {
    background-color: #21232a;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    padding: 10px;
    font-size: 1rem;
    margin: 0 2.5px;
}
