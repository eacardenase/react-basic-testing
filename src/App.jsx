import { useState } from 'react';
import CounterApp from './CounterApp';

export const App = () => {
    return (
        <>
            <CounterApp value={0} />
        </>
    );
};
