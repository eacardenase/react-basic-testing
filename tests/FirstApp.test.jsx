import { cleanup, render, screen } from '@testing-library/react';
import { describe, test, expect, beforeEach } from 'vitest';

import { FirstApp } from '../src/FirstApp';

describe('<FirstApp />', () => {
    beforeEach(cleanup);

    test('should match with the snapshot', () => {
        const title = 'Hola soy Goku';

        const { container } = render(<FirstApp title={title} />);

        expect(container).toMatchSnapshot();
    });

    test('should show title inside h1 tag', () => {
        const title = 'Hola soy Goku';

        const { container, getByText } = render(<FirstApp title={title} />);

        expect(getByText(title)).toBeTruthy();

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain(title);
    });
});
