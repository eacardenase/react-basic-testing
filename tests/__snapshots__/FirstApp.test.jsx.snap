// Vitest Snapshot v1

exports[`<FirstApp /> > should match with the snapshot 1`] = `
<div>
  <h1>
     
    Hola soy Goku
     
  </h1>
  <p>
    I'm a software developer
  </p>
  <p />
</div>
`;
